x/poe/contract/tg4_engagement_integration_test.go

```diff
@@ -49,3 +53,48 @@ func TestEngagementUpdateAdmin(t *testing.T) {
 	gotErr := engagementContract.UpdateAdmin(ctx, newAddress, systemAdminAddr)
 	require.NoError(t, gotErr)
 }
+
+func TestQueryDelegated(t *testing.T) {
+	// setup contracts and seed some data
+	ctx, example, vals, _ := setupPoEContracts(t)
+	vals = clearTokenAmount(vals)
+
+	contractAddr, err := example.PoEKeeper.GetPoEContractAddress(ctx, types.PoEContractTypeEngagement)
+	require.NoError(t, err)
+
+	specs := map[string]struct {
+		ownerAddr string
+		expVal    contract.DelegatedResponse
+		expError  bool
+	}{
+		"query one validator": {
+			ownerAddr: vals[0].OperatorAddress,
+			expVal:    contract.DelegatedResponse{Delegated: vals[0].OperatorAddress},
+		},
+		"query other validator": {
+			ownerAddr: vals[1].OperatorAddress,
+			expVal:    contract.DelegatedResponse{Delegated: vals[1].OperatorAddress},
+		},
+		"query with invalid address": {
+			ownerAddr: "not an address",
+			expError:  true,
+		},
+	}
```

Comment:
Can you add a test case with an unknown address, so that we have this covered as spec?

Reply:
Done.